crates/ide/src/hover.rs

```diff
@@ -272,6 +274,24 @@ fn hover_markup(
     }
 }
 
+fn process_markup(
+    db: &RootDatabase,
+    def: Definition,
+    markup: &Markup,
+    links_in_hover: bool,
+    markdown: bool,
+) -> Markup {
+    let markup = markup.as_str();
+    let markup = if !markdown {
+        remove_markdown(markup)
+    } else if links_in_hover {
+        rewrite_links(db, markup, &def)
+    } else {
+        remove_links(markup)
+    };
+    Markup::from(markup)
+}
```

Comment:
I also feel that our docs pipeline is in the need of refactor. In particular, markdown removal really should be a methond on markap, called from an ide:

```rust
// in handlers.rs

if snap.config.caps.support_markdown() {
  markup.as_markdown()
} else {
  markup.as_plain_text()
}
```

bors r+

Reply:
See #4326 and #6576